content/articles/searching-and-sorting-in-matlab/index.md

```diff
@@ -0,0 +1,162 @@
+```matlab
+else
+index = binary_search_recursive(vector,target,mid+1, last);
+end
+```
+Finally, we call our function to search for the values after the mid element and the result given in the output. 
+This method is referred to as binary. It's because it constantly divides the list into two. For example, if you want to search a value in the range 1-100, the search ranges will be 100, 50, 25, 12, 6, 3, 1. It means we will have seven calls of our function to find the target. If we look at this, we see that the binary search has carried less than 40 comparisons to reach the target compared to the sequential search, which could have taken 100 comparisons if the target is not within the range. 
+
+### Example
+We are given vector A = [2 13 17 22 29 43 64 73 82 89 99] and we want to locate 89 using our function.
+```matlab
+binary_search_recursive(A, 89, 1, length(A))
+binary_search_recursive(A, 88, 1, length(A))  %unavailable 
+```
+![output for target and out of range](/engineering-education/searching-and-sorting-using-matlab/search_two.png)
+
+In a duplicate value, the function gives the index of the value it first locates, unlike a sequential search that finds the first variable in the list.
+### Iterative implementation of binary search
+Here, the first and the last variables are set to be the first and the last indices of the range, respectively. The division of the list into smaller parts involves changing either the first or the last variable one at a time. It is done by moving the first towards the end and the last variable towards the first. It is made possible by the use of the `while`  loop. This while loop is halted when the first value is greater than the last, which means that the target is not found.
+```matlab
+function index = binary_search_iterative(vector,target,first,last)
+found = false;
+while first <= last && ~found
+mid = fix( (first + last) /2 );
+if target < vector(mid)
+last = mid - 1;                   %moving towards the beginning
+elseif target > vector(mid)
+first = mid + 1 ;                 %moving towards the end
+else
+found = true;
+end
+end
+if found
+index = mid;                      %Return the index if found
+else
+index = -1;                       % Return -1 if not found
+end
+```
+Incase we have a vector A = [2 13 17 22 29 43 64 73 82 89 99] and we want to locate 73, the the call of this function will be
+```matlab
+binary_search_iterative(A, 73, 1, length(A))
+```
+The output will be;
+![index of 73](/engineering-education/searching-and-sorting-using-matlab/search_three.png)
+
+### Sorting
+We must know how to sort our data since we have seen the advantage of binary search over-sequential search. As we have seen before, binary search deals with sorted data. Generally, we have three types of sorts: selection sort, quick sort, and merge sort. First, we will look at quicksort and merge sort.
+
+### 1. Quicksort
+It is the most efficient type of sort and also easy to understand.
```

Comment:
I find this a bit loose and ambiguous, please give a reader a clear description of quick sort. Just briefly.

Reply:
Also before you proceed to code sample, introduce a reader on what you are trying to do or achieve.